aten/src/ATen/native/vulkan/VulkanConvolution.cpp

```diff
@@ -66,14 +66,14 @@ ContextConv2D create(
   const auto stride_expanded = expand_param_if_needed(stride, "stride", 2);
   const auto dilation_expanded =
       expand_param_if_needed(dilation, "dilation", 2);
-  const Tensor weight_nchw = weight.contiguous();
+  Tensor weight_nchw = weight.contiguous();
```

Comment:
Does a call to contiguous() return NCHW format?  This always confuses me and for some reason I always default to the assumption that contiguous() returns a tensor laid out in memory contiguously while the elements can be in any order as long as they are all adjacent without any holes.

Anyhow, do you depend on the memory to be NCHW or do you just want the values to be adjacent in memory?  We have noticed in the past that it is very easy to cause unnecessary memory copies with contiguous() calls so if performance matters, you want to vett all usage of this function to make sure it is only doing the minimal work necessary.

Reply:
Ideally we can call this out explicitly in each contiguous call. Like `contiguous(MemoryFormat::Contiguous)`. Less confusion.